components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java

```diff
@@ -42,17 +47,14 @@ public UserSessionDAOImpl() {
 
     public UserSession getSession(String sessionId) throws SessionManagementServerException {
 
-        List<Application> applicationList;
         HashMap<String, String> propertiesMap = new HashMap<>();
-        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate(JdbcUtils.Database.SESSION);
 
         try {
-            applicationList = jdbcTemplate.executeQuery(SQLQueries.SQL_GET_APPLICATION, (resultSet, rowNumber) ->
-                            new Application(resultSet.getString(1),
-                                    resultSet.getString(2),
-                                    resultSet.getString(3),
-                                    resultSet.getString(4)),
-                    preparedStatement -> preparedStatement.setString(1, sessionId));
+            List<Application> applicationList = getApplicationForSessionID(sessionId);
+            for (Application application : applicationList) {
+                getApplicationFromAppID(application);
```

Comment:
Since this will issue multiple queries for the applications in the session, can we do some improvement to reduce the number of queries?